src/preset_cli/cli/superset/sync/native/command.py

```diff
@@ -310,6 +320,19 @@ def get_charts_uuids(config: AssetConfig) -> Iterator[str]:
             yield child["meta"]["uuid"]
 
 
+def get_dataset_filter_uuids(config: AssetConfig) -> Set[str]:
+    """
+    Extract dataset UUID for datasets that are used in dashboard filters.
+    """
+    dataset_uuids = set()
+    for filter_config in config["metadata"].get("native_filter_configuration", []):
+        for target in filter_config["targets"]:
+            if uuid := target.get("datasetUuid"):
```

Comment:
From my tests:
* Even if the dashboard doesn't have any elements, it always have the `position` and `metadata` keys/configurations.
* If the dashboard never had a filter created, it might not include the `native_filter_configuration` key.
* Even filters that don't have a target (such as temporal) still include a blank `targets` key.
* If the filter dataset was deleted, its `target` config won't have a `datasetUuid` field.